src/oop/programs_with_multiple_classes/medium/q6/MainTest.java

```diff
@@ -0,0 +1,97 @@
+import java.util.stream.Stream;
+
+import static global.tools.CustomAssertions.*;
+import static org.junit.jupiter.api.Assertions.assertNull;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+public class MainTest {
+    // Java
+
+    private final String meatClass = "Meat";
+    private final String fridgeClass = "Fridge";
+    private final String varSize = "size";
+    private final String varSpoiled = "isSpoiled";
+    private final String varMeat = "meats";
+    private final String methodName = "timePassed";
+    public ObjectTest meat;
+    public ObjectTest fridge;
+
+    @BeforeEach
+    public void setup() {
+        String meatClassString = "oop.programs_with_multiple_classes.medium.q6." + meatClass;
+        String fridgeClassString = "oop.programs_with_multiple_classes.medium.q6." + fridgeClass;
+        meat = new ObjectTest(meatClassString);
+        fridge = new ObjectTest(fridgeClassString);
+        String modifiedVegMessage =
+                "You have modified the class fields in the " + meatClass + " class. Please revert them back to the original state they were provided in.";
+        assertTrue(meat.hasField(varSpoiled, boolean.class), modifiedVegMessage);
+        assertTrue(meat.hasModifier(varSpoiled, "private"), modifiedVegMessage);
+        Class<?>[] classArguments = {boolean.class};
+        assertTrue(meat.hasConstructor(classArguments), modifiedVegMessage);
+        assertTrue(meat.hasModifier(classArguments, "public"), modifiedVegMessage);
+        String modifiedCupMessage =
+                "You have modified the class fields in the " + fridgeClass + " class. Please revert them back to the original state they were provided in.";
+        assertTrue(fridge.hasField(varSize, int.class), modifiedCupMessage);
+        assertTrue(fridge.hasModifier(varSize, "private"), modifiedCupMessage);
+        assertTrue(fridge.hasField(varMeat, Meat[].class), modifiedCupMessage);
+        assertTrue(fridge.hasModifier(varMeat, "private"), modifiedCupMessage);
+    }
+
+    @Test
+    public void fridgeClassHasRequiredConstructor() {
+        Class<?>[] classArguments = {int.class};
+        assertTrue(fridge.hasConstructor(classArguments),
+                "Your " + fridgeClass + " constructor does not have the correct parameters.");
+        assertTrue(fridge.hasModifier(classArguments, "public"),
+                "Your " + fridgeClass + " constructor does not have the correct visibility modifier.");
+    }
+
+    private static Stream<Boolean> meatInputProvider() {
+        return Stream.of(
+                true, false
+        );
+    }
+
+    @ParameterizedTest
+    @MethodSource("meatInputProvider")
+    public void correctMeatTimePassedMethod(boolean isSpoiled) throws Throwable {
+        Object[][] arguments = {
+                {isSpoiled, boolean.class}
+        };
+        Object vegetableInstance = meat.createInstance(arguments);
+        Object output = meat.callMethod(methodName, vegetableInstance);
```

Comment:
Also, please put the call method after the checks for the method.

Reply:
And you should check if the method exists before we try calling it for better feedback to the student.